### PR TITLE
Fix incorrect clip length when wrapping IAB (#47)

### PR DESCRIPTION
**Pull request by palemieux**

### CODE REVIEW

File: src/AS_02_IAB.cpp [hunk 1]
```diff
@@ -272,7 +272,7 @@ AS_02::IAB::MXFWriter::Finalize() {
 
     byte_t clip_buffer[8] = { 0 };
 
-    ui64_t size = static_cast<ui64_t>(this->m_Writer->m_StreamOffset);
+    ui64_t size = static_cast<ui64_t>(this->m_Writer->m_StreamOffset) /* total size of the KLV */ - 24 /* K and L length */;
```
Comment (kblinova) [inline src/AS_02_IAB.cpp:275]: Can the payload have a different offset, not just 24?

File: src/AS_02_IAB.cpp [hunk 2]
```diff
@@ -272,7 +272,7 @@ AS_02::IAB::MXFWriter::Finalize() {
 
     byte_t clip_buffer[8] = { 0 };
```
Comment (kblinova) [inline src/AS_02_IAB.cpp:273]: The section between /* write clip length */ and /* write footer */ looks like FinalizeClip() function. Is there any reason you are not using it?

Comment (palemieux) [reply src/AS_02_IAB.cpp:273]: Do you mean https://github.com/cinecert/asdcplib/blob/58628942da20e483f5c3076d6c14ff4bef71fc71/src/h__02_Writer.cpp#L356 ?

If so, that function assumes a fixed frame size if I am not mistaken.

File: src/AS_02_IAB.cpp (hunk 1 above)
Comment (palemieux) [reply src/AS_02_IAB.cpp:275]: 24 is hard-coded at https://github.com/cinecert/asdcplib/blob/38daae206397d601738df407df15d1bf1caf01ae/src/AS_02_IAB.cpp#L180 (16 bytes for the UL and 8 bytes for the BER length)

This should probably be turned into a constant...

File: src/AS_02_IAB.cpp (hunk 2 above)
Comment (kblinova) [reply src/AS_02_IAB.cpp:273]: Ah, yes, correct. Nevermind.

File: src/AS_02_IAB.cpp (hunk 1 above)
Comment (kblinova) [reply src/AS_02_IAB.cpp:275]: Yes, those 24s are all over the place...

Comment (palemieux) [reply src/AS_02_IAB.cpp]: @kblinova See revised source.

**Pull request merged**